### ISSUE TITLE
Roundcube fails to retrieve emails from Office 365 [bug] 

### ISSUE THREAD

**Issue by scotlad9** [YetiForceCompany/YetiForceCRM#14489]
### ⁉️ Describe the bug

Roundcube times out when trying to retrieve emails from Office 365.  If the order (under options) is changed to 'descending' then the 30 oldest emails are displayed almost immediately. 

### 🔥 How to trigger the error

Configure IMAP and SMTP settings for Yetiforce Roundcube, login and try to view inbox with default settings

### 👎 Actual Behavior

'Loading' displays and then a timeout error appears.  No emails are visible in the inbox but the folder list correctly displays the number of unread emails in each folder.  Using the 'options' to change the order from 'descending' to 'ascending' retrieves the oldest 30 emails (out of about 200) in the inbox.  Clicking the forward or backward buttons causes the loading and timeout messages to appear again and no emails are displayed.

### 👍 Expected Behavior

A fresh stand-alone installation of Roundcube webmail on the same server works correctly with the same mail account and connection settings used in Yetiforce. 

### 👷 Testing

I am using Version 6.1 with service pack 7 to evaluate Yetiforce CRM.

### 🌍 Your Environment

| Environment                         | Version / Name |
| ----------------------------------- | -------------- |
| YetiForce                       |  6.1 |
| Web server (name and version)       | Apache2  V2.4.41               |
| PHP                                 |    V7.4           |
| Browser (name and version)          | MS Edge or Firefox                |
| Operating System (name and version) |  Linux Mint 20.1              |
| Database (name and version)         |    MariaDB 10.3.25            |
| Service Pack version                |  7               |

### ❓ Additional context

Have searched Yetiforce and Rouncube discussions without finding a solution.  Don't understand why Roundcube works correctly on the same server with the same options and settings.

**Comment by mariuszkrzaczkowski** [comment]
log show and try to disable the yetiforce plugin

**Comment by scotlad9** [comment]
Thankyou for you prompt reply.  I disabled the plugin by renaming the yetiforce.php file in the Roundcube plugins folder. logged back in to YetiForce and Roundcube correctly displayed the mailboxes and allowed me to page forward and back through the inbox.
I'm not sure which log files you need and how to enable them - is it by switching on debug mode?

**Comment by mariuszkrzaczkowski** [comment]
In this file you can disable plugins in Roundcube

config\Modules\OSSMail.php
```php
/** List of plugins */
public static $plugins = [
	'thunderbird_labels',
	'zipdownload',
	'archive',
	'newmail_notifier',
	'advanced_search',
	'contextmenu',
	'identity_smtp',
	//'yetiforce',
];
```
Debugging https://yetiforce.com/pl/component/zoo/item/debugging

Logs are activated in the file

config\Debug.php
https://github.com/YetiForceCompany/YetiForceCRM/blob/ef0e76a6765e0b5bcf51830b39d7a2fd90bbf97e/config/Debug.php#L20-L33

https://github.com/YetiForceCompany/YetiForceCRM/blob/ef0e76a6765e0b5bcf51830b39d7a2fd90bbf97e/config/Debug.php#L111-L145

**Comment by mariuszkrzaczkowski** [comment]
What version of Rouncube do you have?
See the public_html/modules/OSSMail/roundcube/version.php

**Comment by mariuszkrzaczkowski** [comment]
In the latest version, you can enable RBL integration which slows down the loading of e-mails

https://doc.yetiforce.com/code/classes/Config-Components-Mail.html#property_rcListCheckRbl
config\Components\Mail.php
```php
	/** Should the message sender on the mail list be verified in the mail client? */
	public static $rcListCheckRbl = false;

```

**Comment by scotlad9** [comment]
Hi Mariusz

Roundcube version is:-

<?php

return [
               'patch' => '2020.12.01',
               'version' => '0.1.5'
];

**Comment by scotlad9** [comment]
Hi Mariusz

I can confirm that Roundcube fetches emails correctly with the plugin disabled.  When it is enabled, only the oldest 30 emails are retrieved when the order is set to ‘ascending’.  Trying to view any others results in a timeout message.  When the order is set to descending (which I guess is what most people want), no emails are displayed and a timeout message eventually appears.

I’ve attached the imap.log and system.log files.

Hope this helps.

David

**Comment by mariuszkrzaczkowski** [comment]
https://github.com/YetiForceCompany/YetiForceCRM/discussions/13965#discussioncomment-253551

**Comment by scotlad9** [comment]
Hi
/** Should the message sender on the mail list be verified in the mail client? */
	public static $rcListCheckRbl = false;

I've added this to Mail.php but Roundcube issue remains when the plugin is enabled.

**Comment by mariuszkrzaczkowski** [comment]
it will be available from version 6.2

**Comment by scotlad9** [comment]
Thanks Mariusz, that would be good.
